hail_search/hail_search_query.py

```diff
@@ -998,23 +1029,133 @@ def _format_results(self, ht, annotation_fields):
+                self._ht.start_locus.contig == self._ht.end_locus.contig,
+                interval.overlaps(hl.interval(self._ht.start_locus, self._ht.end_locus)),
+                interval.contains(self._ht.start_locus) | interval.contains(self._ht.end_locus),
+            ))
+            if exclude_intervals:
+                interval_filter = ~interval_filter
+            self._ht = self._ht.filter(interval_filter)
+
+        return super()._filter_annotated_table(*args, **kwargs)
+
+    @staticmethod
+    def get_x_chrom_filter(ht, x_interval):
+        return x_interval.contains(ht.start_locus)
+
+    def _get_family_passes_quality_filter(self, quality_filter, annotations=None, **kwargs):
+        passes_quality = super()._get_family_passes_quality_filter(quality_filter)
+        if not annotations.get(NEW_SV_FIELD):
+            return passes_quality
+
+        entries_has_new_call = lambda entries: entries.any(lambda x: x.concordance.new_call)
+        if passes_quality is None:
+            return entries_has_new_call
+
+        return lambda entries: entries_has_new_call(entries) & passes_quality(entries)
+
+    def _get_allowed_consequences_annotations(self, annotations, annotation_filters, is_secondary=False):
+        if is_secondary:
+            # SV search can specify secondary SV types, as well as secondary consequences
+            annotation_filters = self._get_annotation_override_filters(annotations)
+        return super()._get_allowed_consequences_annotations(annotations, annotation_filters)
+
+    def _get_consequence_filter(self, allowed_consequence_ids, annotation_exprs):
+        return self._ht.sorted_gene_consequences.any(
+            lambda gc: allowed_consequence_ids.contains(gc.major_consequence_id)
+        )
+
+    def _get_annotation_override_filters(self, annotations, **kwargs):
+        annotation_filters = []
+        if annotations.get(STRUCTURAL_ANNOTATION_FIELD):
+            allowed_type_ids = self._get_enum_terms_ids('sv_type', None, annotations[STRUCTURAL_ANNOTATION_FIELD])
+            if allowed_type_ids:
+                annotation_filters.append(hl.set(allowed_type_ids).contains(self._ht.sv_type_id))
+
+        return annotation_filters
+
+    def _additional_annotation_fields(self):
+        sv_type_enum = self._enums['sv_type']
+        insertion_type_id = sv_type_enum.index('INS')
+        get_end_chrom = self.BASE_ANNOTATION_FIELDS['endChrom']
+        return {
+            'cpxIntervals': lambda r: self._format_enum(
+                r, 'cpx_intervals', {'type': sv_type_enum}, annotate_value=lambda val, *args: {
+                    'chrom': val.start.contig,
+                    'start': val.start.position,
+                    'end': val.end.position,
+                },
+            ),
+            # For insertions, end_locus represents the svSourceDetail, otherwise represents the endChrom
+            'endChrom': lambda r: hl.or_missing(r.sv_type_id != insertion_type_id, get_end_chrom(r)),
```

Comment:
I had pretty good luck with just defining a hail expression as a variable and re-using:

```
end_chrom = get_end_chrom(r)
...
'endChrom': lambda r: hl.or_missing(r.sv_type_id != insertion_type_id, end_chrom),
'svSourceDetail': lambda r: hl.or_missing(r.sv_type_id == insertion_type_id, hl.or_missing(hl.is_defined(end_chrom), hl.struct(chrom=end_chrom)))
```

Reply:
the problem is `r` is not defined above where you do `end_chrom = get_end_chrom(r)`, its only available in the lambda